autobot/api.py

```diff
@@ -8,77 +8,73 @@
 
 """Autobot API."""
 
+import copy
+
 import yaml
 
 from autobot.config_loader import Config
 from autobot.github import GitHubAPI
 
 
 class BotAPI:
-    """Generates report."""
+    """Bot's API."""
 
-    @classmethod
-    def __init__(cls, config: Config):
+    def __init__(self, config: Config):
         """Bot initialization."""
-        cls.config = config.config
-        cls.report = GitHubAPI(
-            cls.config["AUTOBOT_OWNER"], cls.config["AUTOBOT_GH_TOKEN"]
-        ).report(cls.load_repositories())
+        self.config = config
 
     @classmethod
-    def generate_report(cls, maintainer: str) -> dict:
-        """Generates a report for a maintainer."""
-        res = cls.report[0]["repos"]
-        print(res)
-        print(
-            "---------------------------------------------------------------------------------"
-        )
-        return res
+    def md_report(cls, report):
+        """Returns the report in markdown format."""
+        lines = []
+        for repo in report.keys():
+            repo_report = report[repo]
+            lines.append(
+                f"\n### [{repo_report['url'].split('/')[-1]}]({repo_report['url']})"
+            )
+            for (action, targets) in repo_report["actions"].items():
+                lines.append(f"- **{action}**")
+                for target in targets:
+                    lines.append(
+                        f"  - {target['url']}: {target['title']} "
+                        f"({target['creation_date'].date()})"
+                    )
+        return "\n".join(lines)
 
-    @classmethod
-    def send_report(cls, maintainer: str, format: str):
-        """Send the report to a maintainer (on Gitter or via email)."""
-        res = cls.generate_report(maintainer)
-        if format == "markdown":
-            """ TO DO """
+    def generate_report(self, **kw):
+        """Returns the report in markdown format."""
+        report = GitHubAPI(
+            self.config["AUTOBOT_OWNER"],
+            self.config["AUTOBOT_GH_TOKEN"],
+            self.config.repositories,
+        ).report()
+        maintainer = kw.get("maintainer", None)
+        if not maintainer:
+            return report
+        maintainer_report = copy.deepcopy(report)
+        for repo in report:
+            repo_report = report[repo]
+            if maintainer not in repo_report["maintainers"]:
+                del maintainer_report[repo]
+        return maintainer_report
 
-    @classmethod
-    def load_repositories_yml(cls):
-        """Load repository.yml file."""
-        return yaml.load(open(cls.config["AUTOBOT_INFO_PATH"]))["orgs"][
-            cls.config["AUTOBOT_OWNER"]
-        ]
+    def formatted_report(self, format):
```

Comment:
```suggestion
    def format_report(self, format):
```

It's better to use verbs for methods/functions